src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -193,11 +200,31 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                         return;
                     }
 
-                    IEnumerable<ConfigurationClient> availableClients = _configClientManager.GetAvailableClients();
+                    IEnumerable<ConfigurationClient> clients = _configClientManager.GetClients();
+
+                    //
+                    // Filter clients based on their backoff status
+                    clients = clients.Where(client => 
+                    {
+                        Uri endpoint = _configClientManager.GetEndpointForClient(client);
+
+                        if (!_configClientBackoffs.TryGetValue(endpoint, out ConfigurationClientBackoffStatus clientBackoffStatus))
+                        {
+                            UpdateClientBackoffStatus(endpoint, true);
+
+                            clientBackoffStatus = _configClientBackoffs[endpoint];
+                        }
+
+                        return clientBackoffStatus?.BackoffEndTime <= DateTimeOffset.UtcNow;
```

Comment:
Regarding the null check (`?`). I see that we either get existing backoff status or create a new one. How could it be null?

Reply:
`DateTimeOffset.UtcNow`, should calculate once outside of the loop.

Reply:
Right it shouldn't be null, leftover from old change